include/linux/hypercall.h

```diff
@@ -4,82 +4,88 @@
 
 static inline void igloo_hypercall(unsigned long num, unsigned long arg1) {
 #if defined(CONFIG_MIPS)
-    register unsigned long a0 asm("a0") = num;
-    register unsigned long a1 asm("a1") = arg1;
+    register unsigned long v0 asm("v0") = num;
+    register unsigned long a0 asm("a0") = arg1;
 
     asm volatile(
        "movz $0, $0, $0"
-        : "+r"(a0)  // Input and output in R0
-        : "r"(a1) // arg1 in register A1
+        : "+r"(v0)  // Input and output in R0
+        : "r"(a0) // num in register v0
         : // No clobber
     );
 
 
-#elif defined(CONFIG_ARM)
-  register uint32_t r0 asm("r0") = num;
-  register uint32_t r1 asm("r1") = arg1;
-  asm volatile(
-     "mov r0, %0 \t\n\
-      mov r1, %1 \t\n\
-      mcr p7, 0, r0, c0, c0, 0"
+#elif defined(CONFIG_ARM) || defined(CONFIG_ARM64)
+  #if defined(CONFIG_ARM64)
+    register uint64_t reg0 asm("x8") = num;
+    register uint64_t reg1 asm("x0") = arg1;
+    asm volatile(
+            "msr S0_0_c5_c0_0, xzr \n"
+            :
+            : "r"(reg0), "r"(reg1)
```

Comment:
Short answer is it doesn't matter right. We don't return anything on the single input version. It's inconsistent but doesn't actually change anything right now. -- Something I'll look into with the generic.

Reply:
I think that's probably okay in the short term then if this will be fixed soon, but I believe there's definitely some risk of unexpected behavior if we get this wrong. For exmaple, if PANDA writes a value into reg0, the compiler wouldn't be expecting it and I think things could go off the rails - i.e., if the compiler has written 1 into reg0, panda writes 2, then a subsequent instruction needs a register containing 1, it might just use reg0 thinking there's no way it could have changed.